webviz_subsurface/private_containers/_surface_selector.py

```diff
@@ -0,0 +1,589 @@
+            if not sens_cases:
+                return [], None, self.hide_dropdown_style
+            value = current_value if current_value in sens_cases else sens_cases[0]
+            options = [{"value": sens, "label": sens} for sens in sens_cases]
+            return options, value, self.show_dropdown_style
+
+        @app.callback(
+            Output(self.storage_id, "children"),
+            [
+                Input(self.attr_id, "value"),
+                Input(self.name_id, "value"),
+                Input(self.date_id, "value"),
+                Input(self.ensemble_id, "value"),
+                Input(self.aggreal_id, "value"),
+                Input(self.realization_id, "value"),
+                Input(self.sens_name_id, "value"),
+                Input(self.sens_case_id, "value"),
+            ],
+        )
+        def _set_data(
+            attr, name, date, ensemble, aggreal, calculation, sens_name, sens_case
+        ):
+
+            """
+            Stores current selections to dcc.Store. The information can be retrieved as a json string
+            from a dash callback Input. E.g. [Input(surfselector.storage_id, 'children')]
+            """
+            reals = self.realizations(ensemble, sens_name, sens_case)
+            all_senscases = [
+                {
+                    "case": senscase,
+                    "realizations": self.realizations(ensemble, sens_name, senscase),
+                }
+                for senscase in self.sens_cases(ensemble, sens_name)
+            ]
+            return json.dumps(
+                {
+                    "attribute": attr,
+                    "name": name,
+                    "date": date,
+                    "ensemble": ensemble,
+                    "aggregation": calculation if aggreal == "Aggregation" else None,
+                    "realization": reals if aggreal == "Aggregation" else calculation,
+                    "sensname": sens_name,
+                    "senscase": sens_case,
+                    "all_senscases": all_senscases,
+                }
+            )
+
+
+def prev_value(current_value, options):
+    try:
+        index = options.index(current_value)
+    except ValueError:
+        index = None
+    if index > 0:
+        return options[index - 1]
+    else:
+        return current_value
+
+
+def next_value(current_value, options):
+    try:
+        index = options.index(current_value)
+    except ValueError:
+        index = None
+    if index < len(options) - 1:
+        return options[index + 1]
+    else:
+        return current_value
+
+
+def format_date(date_string):
+    """Reformat date string for presentation
+    20010101 => Jan 2001
+    20010101_20010601 => (Jan 2001) - (June 2001)
+    20010101_20010106 => (01 Jan 2001) - (06 Jan 2001)"""
+    d = str(date_string)
+    if len(d) == 8:
+        d = datetime.strptime(d, "%Y%m%d").strftime("%b %Y")
+    if len(d) == 17:
+        begin = d[0:8]
+        end = d[9:17]
```

Comment:
Instead `[begin, end] = [datetime.strptime(date, '%Y%m%d') for date in d.split('-')]`?